### ISSUE TITLE
Fix Age Rage Preference query logic on Child Matching dashboard

### ISSUE THREAD

**Issue by n-devr** [mareinc/mare#302]
The current logic is not returning all families with Preferred Age Ranges that intersect with the range defined on the dashboard.

**Example**

The specified age range is 4-9, four families are returned as potential matches:

![image.png](https://images.zenhubusercontent.com/585873f3b9dc3c622023c5fd/4a849a3c-cc23-423c-9137-412508f851ef)

After adjusting the age range to 3-9, two new families are returned with age ranges from 3-6.  Those families should have matched the initial 4-9 range (ages 4, 5, and 6 were shared between both ranges).

![image.png](https://images.zenhubusercontent.com/585873f3b9dc3c622023c5fd/083264e2-b89b-4d87-bf1b-0dc471287f77)

**Comment by n-devr** [comment]
Question for @joesmare - is `0` a valid value in the TO dropdown for age range?  It seems like that would never return results and could be removed.

**Comment by joesmare** [comment]
@n-devr if someone were searching for an infant, under the age of 1 then you could potentially set a range of 0 to 0 for that search

**Comment by n-devr** [comment]
As I'm coding this solution I'm noticing that we have to put guards and define 'smart' defaults in the case that a family doesn't have a preferred age range set.  We could avoid this and implement a cleaner, more consistent, solution by making those fields required in the Family model, or by defining default values for them.  Is that something that we could look into?

@joesmare @bchiaruttini

**Comment by joesmare** [comment]
@n-devr your suggestion makes sense, with a caveat with regard to when it is made mandatory. We could make the necessary fields mandatory...but only at the point where the 'Registered with MARE' box is checked and prevent a save immediately after that action is taken. This would include the following fields:

1. Current Children in Family: Number of Children
2. Matching Preferences: From Age
3. Matching Preferences: To Age
4. Matching Preferences: Minimum Number of Children
5. Matching Preferences: Maximum Number of Children

This extends beyond your initial questions, as related to age only, but these are all fields required for reporting and to avoid null values.

**Comment by n-devr** [comment]
OK - given the difficulty of making fields required only after certain conditions have been met I think we should try to tackle this with smart defaults, or perhaps it's not worth changing at all.  Having to contend with unset or missing values will add complexity to the queries made from the reporting dashboard, but that may be a necessary evil.

We should discuss in more detail when we meet next.  It may be worth having a meeting about this with a wider group as well, as the decision we make here will impact functionality across reporting dashboards.

**Comment by n-devr** [comment]
I ended up writing a query that could handle the missing fields, let's just touch base to ensure there aren't other reasons to move towards implementing smart defaults for the fields you listed.  It's certainly still an option and would make the query logic simpler and improve the display of the reporting dashboard, however it would also introduce risk to any exisitng code that isn't expecting default values.

**Comment by joesmare** [comment]
Closing for age range. Creating a new story to address the missing data issue. Will put on hold for possible future work.
